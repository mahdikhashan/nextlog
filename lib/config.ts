function readConfig(): NextlogConfig {
  try {
    const configModule = require("../nextlog.config");
    return configModule.default || configModule;
  } catch (error) {
    console.error(`Error reading config file: ${error}`);
    throw error;
  }
}

export const DefaultNextlogConfig = {
  theme: "default",
  layout: "list",
} as NextlogConfig;

export function NextlogConfig(config = DefaultNextlogConfig) {
  return config;
}

const config = readConfig();

export default config as NextlogConfig;
