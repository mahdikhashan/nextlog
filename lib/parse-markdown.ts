import matter from "gray-matter";

type ParseMarkdownArgument = {
  md: string;
}

function parseMarkdown({ md }: ParseMarkdownArgument) {
  return matter(md);
}

export default parseMarkdown;
