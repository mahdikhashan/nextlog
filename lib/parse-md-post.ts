import matter from "gray-matter";

function parseMarkdown(md: string) {
  return matter(md);
}

export default parseMarkdown;
