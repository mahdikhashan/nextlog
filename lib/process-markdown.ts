import { GrayMatterFile } from "gray-matter";
import { remark } from "remark";
import html from "remark-html";

type processMarkdownArg = GrayMatterFile<string>;

async function processMarkdown(markdown: processMarkdownArg) {
  const processedContent = await remark()
    .use(html)
    .process(markdown.content);
  const contentHTML = processedContent.toString();

  return contentHTML
}

export default processMarkdown;
