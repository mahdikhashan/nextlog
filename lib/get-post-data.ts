import * as fs from "fs";
import * as path from "path";

import { default as parseMarkdown } from "@/lib/parse-markdown";
import { default as processMarkdown } from "@/lib/process-markdown";

interface GetPostDataArguments {
  id: string;
}

export async function getPostData(args: GetPostDataArguments) {
  const { id } = args;

  const fullPath = path.join(process.cwd(), "content/posts", `${id}.md`);
  const postFile = fs.readFileSync(fullPath, "utf8");

  const post = parseMarkdown({ md: postFile });

  const processedPost = await processMarkdown(post);

  return {
    id,
    content: processedPost,
    ...post.data,
  };
}
