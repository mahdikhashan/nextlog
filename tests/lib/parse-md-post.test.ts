import * as fs from "fs";
import * as path from "path";

import { default as parseMarkdown } from "@/lib/parse-markdown";

test("parses inline markdown correctly", function () {
  const md = "---\nfoo: bar\n---\nThis is content";
  const parsed_markdown = parseMarkdown({ md });
  expect(parsed_markdown).toStrictEqual({
    content: "This is content",
    data: {
      foo: "bar",
    },
    excerpt: "",
    isEmpty: false,
  });
});

test("should have correct properties", function () {
  //TODO test md properties
  expect(true).toBe(true);
})

describe("parses markdown files correctly", function () {
  const fixturesPath = path.join(__dirname, "../fixtures");
  const markdownFiles = fs
    .readdirSync(fixturesPath)
    .filter((file) => file.endsWith(".md"));

  markdownFiles.forEach((markdownFile) => {
    test(`${markdownFile} should match snapshot correctly`, () => {
      const markdownContent = fs.readFileSync(
        path.join(fixturesPath, markdownFile),
        "utf-8"
      );
      const parsedResult = parseMarkdown({ md: markdownContent });
      expect(parsedResult).toMatchSnapshot();
    });

    test(`${markdownFile} should have valid data properties`, () => {})
  });
});
