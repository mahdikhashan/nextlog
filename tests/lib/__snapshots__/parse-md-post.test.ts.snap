// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parses markdown files correctly should parse faulty-post.md correctly 1`] = `
{
  "content": "--+
title: Sample Post Title
---

content of it
",
  "data": {},
  "excerpt": "",
  "isEmpty": false,
}
`;

exports[`parses markdown files correctly should parse sample-post.md correctly 1`] = `
{
  "content": "
# Description

\`\`\`javascript
const app = express();
\`\`\`

## Example Section

1. valid
2. not valid
3. simple error
",
  "data": {
    "date": "01-31-2024",
    "image": "content/posts/nuxtlog-announce.png",
    "title": "Sample Post Title",
  },
  "excerpt": "",
  "isEmpty": false,
}
`;
